dehydrated

```diff
@@ -848,13 +848,26 @@ sign_csr() {
   fi
 
   # Finally request certificate from the acme-server and store it in cert-${timestamp}.pem and link from cert.pem
-  echo " + Requesting certificate..."
   csr64="$( <<<"${csr}" "${OPENSSL}" req -config "${OPENSSL_CNF}" -outform DER | urlbase64)"
   if [[ ${API} -eq 1 ]]; then
+    echo " + Requesting certificate..."
     crt64="$(signed_request "${CA_NEW_CERT}" '{"resource": "new-cert", "csr": "'"${csr64}"'"}' | "${OPENSSL}" base64 -e)"
     crt="$( printf -- '-----BEGIN CERTIFICATE-----\n%s\n-----END CERTIFICATE-----\n' "${crt64}" )"
   else
-    result="$(signed_request "${finalize}" '{"csr": "'"${csr64}"'"}' | clean_json | get_json_string_value certificate)"
+    echo " + Finalizing order..."
+    order="$(echo ${finalize} | sed 's,/*[^/]\+/*$,,')"
+    result="$(signed_request "${finalize}" '{"csr": "'"${csr64}"'"}' | clean_json)"
+    reqstatus="$(printf '%s\n' "${result}" | get_json_string_value status)"
+
+    while [[ "${reqstatus}" = "processing" ]]; do
```

Comment:
I would say you should match `processing` as well.

If the status is `invalid`, what do you do here? just continue and the next request will fail (but good luck debugging)? Maybe test against that value as well?

Something like this maybe? (I haven't tested it)
```sh
while :
do
  case "${reqstatus}"
  in
    "processing" | "pending")
      echo "   > Order in status ${reqstatus}, retrying in a while..."
      sleep 5;
      ;;
    "valid")
      break;
      ;;
    *)
      _exiterr "Order in status ${reqstatus}"
      ;;
  esac
  result="$(signed_request "${order}" '{}' | clean_json)"
  reqstatus="$(printf '%s\n' "${result}" | get_json_string_value status)"
done
```